src/core/ports/commands/get-user-key-results-statuses.command.ts

```diff
@@ -0,0 +1,118 @@
+import { uniqBy } from 'lodash'
+
+import { Status } from '@core/interfaces/status.interface'
+// eslint-disable-next-line import/order
+import { KeyResultCheckIn } from '@core/modules/key-result/check-in/key-result-check-in.orm-entity'
+
+// Export interface Status {
+//     progress: number
+//     confidence: number
+//     isOutdated: boolean
+//     latestCheckIn?: KeyResultCheckInInterface
+//     checkmarks?: KeyResultCheckMarkInterface[]
+//   }
+
+import { KeyResultInterface } from '@core/modules/key-result/interfaces/key-result.interface'
+import { KeyResult } from '@core/modules/key-result/key-result.orm-entity'
+
+import { Command } from './base.command'
+
+type Options = {
+  active?: boolean
+  onlyOwnerKeyResults?: boolean
+}
+
+export class GetUserKeyResultsStatusesCommand extends Command<any[]> {
+  public async execute(
+    userID: string,
+    filters?: KeyResultInterface,
+    options?: Options,
+  ): Promise<KeyResult[]> {
+    const rowsOwnerKeyResults = await this.core.entityManager.query(
+      `
+      WITH latest_check_in_by_kr AS
+            (SELECT DISTINCT ON (kr.id) krci.*
+            FROM public.key_result_check_in krci
+            JOIN public.key_result kr ON krci.key_result_id = kr.id
+            ORDER BY kr.id,
+                      krci.created_at DESC),
+         latest_check_in_by_kr_with_user as (
+          select lci.*, concat('{"fullName": "', u.first_name, u.last_name, '"}')::json as user from latest_check_in_by_kr lci join "user" u on lci.user_id = u.id)
+          
+          select kr.*, to_json(krs.*) as status, to_json(lci.*) as latest_check_in from key_result kr 
+      join key_result_status krs on kr.id = krs.id
```

Comment:
Remember to use explicit joins, in this case, an inner join.
Also, be consistent on upper or lower case through queries.